python/ray/dag/compiled_dag_node.py

```diff
@@ -1235,15 +1272,15 @@ def _get_or_compile(
             # Step 1: populate `arg_to_consumers` and perform some validation.
             for task in tasks:
                 has_at_least_one_channel_input = False
-                for arg in task.args:
```

Comment:
If `task.args` contains only a list of `DAGNode`s, such as `[DAGNode]`, `has_at_least_one_channel_input` will remain False, and a `ValueError` will be raised. Hence, use `_upstream_nodes` instead.